compiler/rustc_hir_analysis/src/check/intrinsic.rs

```diff
@@ -57,9 +55,17 @@ fn equate_intrinsic_type<'tcx>(
         }
     };
 
+    // the host effect param should be invisible as it shouldn't matter
+    // whether effects is enabled for the intrinsic provider crate.
+    let consts_count = if generics.host_effect_index.is_some() {
```

Comment:
This could theoretically be restructured to use `TyCtxt::has_host_param` at the cost of calling `generics_of` twice in total, so eh, dunno.